renaissance-core/src/main/java/org/renaissance/RenaissanceBenchmark.java

```diff
@@ -96,13 +96,19 @@ public final Throwable runBenchmark(Config config) {
       }
       return null;
     } catch (Throwable t) {
+      for (ResultObserver observer : config.resultObservers()) {
+        observer.onFailure(name());
+      }
       return t;
     } finally {
       try {
         tearDownAfterAll(config);
       } catch (Throwable t) {
         System.err.println("Error during tear-down: " + t.getMessage());
         t.printStackTrace();
+        for (ResultObserver observer : config.resultObservers()) {
+          observer.onFailure(name());
```

Comment:
Another thing - should you be passing the exception object to the observer too?

Reply:
Also, should `t` be returned here?

Reply:
> Also, should t be returned here?

I am not sure why we do not return `t` here but it definitely makes sense. Added.

> Another thing - should you be passing the exception object to the observer 

Maybe, it is not needed now.